plugins/bbcode/plugin.js

```diff
@@ -651,7 +665,17 @@
 				}
 			} );
 
+			var entities = {
+				'[': '&#91;',
+				']': '&#93;'
+			};
+
 			editor.dataProcessor.htmlFilter.addRules( {
+				text: function( value ) { // Replace square brackets with entities (#2782).
+					return value.replace( /(\[|\])/g, function( match ) {
+						return entities[ match ];
```

Comment:
I'm wondering if we can use `String.prototype.charCodeAt` here and remove `entities` array altogether:

```suggestion
						return '&#' + match.charCodeAt( 0 ) + ';';
```